src/parsers/json/Parser.cpp

```diff
@@ -482,14 +482,48 @@ const ElemR* Parser::boxToElemGroup(Mesh::Unstructured& mesh, const std::string&
 {
     BoxR3 box = strToBox(line);
     std::unique_ptr<ElemR> elem;
-    if (box.isVolume()) {
-        elem = std::make_unique<HexR8>(mesh.coords(), ElemId(0), box);
+
+    auto posVec = box.getPos();
+    std::vector<const Geometry::CoordR3*> coords;
+
+    if (box.isVolume()) {    
+        for (std::size_t i = 0; i < Geometry::HexR8::sizeOfCoordinates; i++) {
```

Comment:
I would refactor these lines into a single function whitch builds coords taking posVec and the numberOfCoordinates.

It is repeated for volumes, surfaces, etc.

Reply:
Thanks for spotting that. Done.